pinecone/client.go

```diff
@@ -80,14 +207,116 @@ func NewClientBase(in NewClientBaseParams) (*Client, error) {
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//	 }
+//
+//  idx, err := pc.Index(idxs[0].Host) // You can now use idx to interact with your index.
+//  defer idx.Close()
+//
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//   }
+ func (c *Client) Index(host string) (*IndexConnection, error) {
 	return c.IndexWithAdditionalMetadata(host, "", nil)
 }
 
+// IndexWithNamespace creates an IndexConnection to the specified host within the specified namespace.
+//
+// Parameters:
+//   - host: The host URL of your Pinecone index.
+//   - namespace: The target namespace.
+//
+// Returns a pointer to an IndexConnection instance on success. In case of failure, it returns nil and an error.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//    ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//    SourceTag: "your_source_identifier", // optional
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//	 }
+//
+//  idx, err := pc.IndexWithNamespace(idxs[0].Host, <"sample-namespace">) // You can now use idx to interact with your index.
+//  defer idx.Close()
+//
+//  if err != nil {
+//    fmt.Println("Error:", err)
+//   }
 func (c *Client) IndexWithNamespace(host string, namespace string) (*IndexConnection, error) {
 	return c.IndexWithAdditionalMetadata(host, namespace, nil)
 }
 
+// IndexWithAdditionalMetadata creates an IndexConnection to the specified host within the specified namespace,
+// with the addition of custom metadata fields.
+//
+// Parameters:
+//   - host: The host URL of your Pinecone index.
+//   - namespace: The target namespace.
+//   - additionalMetadata: A map of additional metadata fields to include in the API request.
+//
+// Returns a pointer to an IndexConnection instance on success. In case of failure,
+// it returns nil and the error encountered.
+//
+// Example:
+//  ctx := context.Background()
+//
+//  clientParams := pinecone.NewClientParams{
+//    ApiKey:    getEnvVars("PINECONE_API_KEY"),
+//    SourceTag: "your_source_identifier", // optional
+//   }
+//
+//  pc, err := pinecone.NewClient(clientParams)
+//  if err != nil {
+//    log.Fatalf("Failed to create Client: %v", err)
+//   }
+//
+//  idxs, err := pc.ListIndexes(ctx)
+//	if err != nil {
+//	  fmt.Println("Error:", err)
+//	 }
+//
+//  idx, err := pc.IndexWithAdditionalMetadata(
+//                idxs[0].Host,
+//                <"sample-namespace">,
+//                map[string]string{"custom-request-metadata": "custom-metadata-values"}
+//              ) // You can now use idx to interact with your index.
```

Comment:
nit: More indentation woes 😬

Reply:
Yeah my linter doesn't do indentation for comments. Sorry about that!